client/map_client.go

```diff
@@ -65,19 +65,18 @@ func (c *MapClient) GetAndVerifyMapLeaves(ctx context.Context, indexes [][]byte)
 	if err != nil {
 		return nil, status.Errorf(status.Code(err), "map.GetLeaves(): %v", err)
 	}
-	if got, want := len(getResp.MapLeafInclusion), len(indexes); got != want {
-		return nil, status.Errorf(status.Code(err), "got %v leaves, want %v", got, want)
-	}
-	mapRoot, err := c.VerifySignedMapRoot(getResp.GetMapRoot())
+	return c.VerifyMapLeavesResponse(indexes, -1, getResp)
+}
+
+// GetAndVerifyMapLeavesByRevision verifies and returns the requested map leaves at a specific revision.
+func (c *MapClient) GetAndVerifyMapLeavesByRevision(ctx context.Context, revision int64, indexes [][]byte) ([]*trillian.MapLeaf, error) {
+	getResp, err := c.Conn.GetLeavesByRevision(ctx, &trillian.GetMapLeavesByRevisionRequest{
+		MapId:    c.MapID,
+		Index:    indexes,
+		Revision: revision,
+	})
 	if err != nil {
-		return nil, status.Errorf(codes.Internal, "VerifySignedMapRoot(%v): %v", c.MapID, err)
-	}
-	leaves := make([]*trillian.MapLeaf, 0, len(getResp.MapLeafInclusion))
-	for _, m := range getResp.MapLeafInclusion {
-		if err := c.VerifyMapLeafInclusionHash(mapRoot.RootHash, m); err != nil {
-			return nil, status.Errorf(status.Code(err), "map: VerifyMapLeafInclusion(): %v", err)
-		}
-		leaves = append(leaves, m.Leaf)
+		return nil, status.Errorf(status.Code(err), "map.GetLeaves(): %v", err)
```

Comment:
Can we avoid this by making the map.GetLeaves() error better? Then we can just return it directly.